includes/Modules/Thank_With_Google/Web_Tag.php

```diff
@@ -0,0 +1,228 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Thank_With_Google\Web_Tag
+ *
+ * @package   Google\Site_Kit\Modules\Thank_With_Google
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Thank_With_Google;
+
+use Google\Site_Kit\Core\Modules\Tags\Module_Web_Tag;
+use Google\Site_Kit\Core\Util\Method_Proxy_Trait;
+use Google\Site_Kit\Core\Tags\Tag_With_DNS_Prefetch_Trait;
+
+/**
+ * Class for Web tag.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Web_Tag extends Module_Web_Tag {
+
+	use Method_Proxy_Trait, Tag_With_DNS_Prefetch_Trait;
+
+	/**
+	 * Publication ID.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	private $publication_id;
+
+	/**
+	 * Button placement.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	private $button_placement;
+
+	/**
+	 * Button post types.
+	 *
+	 * @since n.e.x.t
+	 * @var string[]
+	 */
+	private $button_post_types;
+
+	/**
+	 * Sets the current publication ID.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $publication_id Publication ID.
+	 */
+	public function set_publication_id( $publication_id ) {
+		$this->publication_id = $publication_id;
+	}
```

Comment:
This method is unnecessary. The publication ID for this tag is the "main" ID to pass into the constructor, so it should rather be looked up from the property in the parent class. It's a bit redundant to set it here too.

Reply:
Good catch. I have created a follow up PR to address this: #5568 5616